src/parameters.rs

```diff
@@ -0,0 +1,357 @@
+    #[value(name = "local-file")]
+    LocalFile,
+}
+
+#[derive(Args, Debug, Clone)]
+pub struct OutputDestinationArgs {
+    /// Platform for the output of the generated key or resource.
+    #[arg(long, value_enum)]
+    output_platform: OutputPlatformChoice,
+
+    /// Folder path where the private key output will be saved (required if --output-platform=local-file).
+    #[arg(long)]
+    output_local_path: Option<PathBuf>,
+}
+
+pub fn create_metadata_for_token_retrieve(
+    client_id: String,
+    environment: Environments,
+    auth_args: &AuthInputArgs,
+    endpoints: ExternalEndpoints,
+) -> Result<SystemTokenCreationMetadata, Box<dyn std::error::Error>> {
+    let auth_method = select_auth_method(
+        auth_args.client_secret.clone(),
+        auth_args.private_key_path.clone(),
+    )?;
+    let environment = select_environment(
+        environment,
+        endpoints.token_renewal_endpoint,
+        endpoints.system_identity_creation_endpoint,
+    )?;
+
+    Ok(SystemTokenCreationMetadata {
+        client_id,
+        environment,
+        auth_method,
+    })
+}
+pub fn create_metadata_for_identity_creation(
+    identity_type: &IdentityType,
+) -> Result<SystemIdentityCreationMetadata, Box<dyn std::error::Error>> {
+    let (basic_auth_args, output_platform) = match identity_type {
+        IdentityType::Secret(secret_args) => (
+            secret_args.basic_auth_args.clone(),
+            OutputPlatform::LocalPrivateKeyPath("./".into()),
+        ),
+        IdentityType::Key(key_args) => (
+            key_args.basic_auth_args.clone(),
+            select_output_platform(
+                &key_args.output_options.output_platform,
+                key_args.output_options.output_local_path.clone(),
+            )?,
+        ),
+    };
+
+    let env = select_environment(
+        basic_auth_args.environment.clone(),
+        Some(
+            basic_auth_args
+                .endpoints
+                .token_renewal_endpoint
+                .unwrap_or_default(),
+        ),
+        Some(
+            basic_auth_args
+                .endpoints
+                .system_identity_creation_endpoint
+                .unwrap_or_default(),
+        ),
+    )?;
+
+    Ok(SystemIdentityCreationMetadata {
+        system_identity_input: SystemIdentityInput {
+            client_id: basic_auth_args.client_id,
+            organization_id: basic_auth_args.organization_id,
+        },
+        name: basic_auth_args.name.clone(),
+        environment: env,
+        output_platform,
+    })
+}
+
+pub fn select_environment(
+    environment: Environments,
+    token_url: Option<String>,
+    identity_url: Option<String>,
+) -> Result<NewRelicEnvironment, Error> {
+    match environment {
+        Environments::US => Ok(NewRelicEnvironment::US),
+        Environments::EU => Ok(NewRelicEnvironment::EU),
+        Environments::Staging => Ok(NewRelicEnvironment::Staging),
+        Environments::Custom => {
```

Comment:
I still do not fully understand why we need this. Is there a test leveraging it? Otherwise I would avoid adding unused code